### DISCUSSION TITLE
Difxapp does not run DefaultInstall section of the INF

### DISCUSSION THREAD

**Discussion by AbsolutelyWrong** [wixtoolset/issues#6650] [Questions] [answered]
## Triage

Issues are triaged at [online meetings](https://www.firegiant.com/blog/), generally held alternate Thursdays at 9:30 a.m. Pacific time (UTC-7/UTC-8). Meeting requests are sent to the [wix-devs](http://wixtoolset.org/documentation/mailinglist/#wix-devs) and wix-users mailing lists, as well as in Discussions.

## Discussions and support

Note that discussions are best held on the mailing lists not in issues. Please do not open issues requesting support or debugging help; first start a thread on the [wix-users](http://wixtoolset.org/documentation/mailinglist/#wix-users) mailing list and open an issue if the discussion suggests a bug in the WiX Toolset or that a feature request is appropriate. You can also try out the beta [Discussions](https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions) feature.

## Instructions

Read the sections above. Delete the Triage, Discussions and support, and Instructions sections and fill out the rest.

## Bugs

If this issue is a bug:

* Which version of WiX are you building with?

> 3.11.2.4516

* Which version of Visual Studio are you building with (if any)?

> Microsoft Visual Studio Professional 2019 Version 16.11.3

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> WiX Toolset Visual Studio Extension version 1.0.0.4

* Which version of .NET are you building with?

> .NET Framework 4.6.1

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> at least Window 10 1903 and newer 

* Describe the problem and the steps to reproduce it.

> I'm trying to install kernel driver using difxapp. Driver is fully certified, .inf, .cat and .sys files are all in respective component.
I tried different combinations of Legacy and ForceInstall flags and still driver would not install.
In setupapi.dev.log I see that driver is stored at driver store, but DefaultInstall section of the INF is never executed (setupapi.dev.log does not contain Plug and Play Service section).
In my case DefaultInstall section registers kernel service and makes changes to registry.
All driver files are OK, since "RMB .INF file -> install" installs driver correctly.

* Describe the behavior you expected and how it differed from the actual behavior.

> I expect difxapp:Driver to execute DefaultInstall section of the INF file after putting driver in driver store.

**Comment by barnson** [comment, marked as answer]
DifxApp has been deprecated for quite a while. WiX doesn't own DifxApp so can't fix any of its bugs.

**Comment by wmanning** [comment]
https://docs.microsoft.com/en-us/windows-hardware/drivers/install/difx-guidelines

**Comment by AbsolutelyWrong** [comment]
Thanks for clarifying

**Comment by AbsolutelyWrong** [comment]
Could you please update
https://wixtoolset.org/documentation/manual/v3/xsd/difxapp/driver.html page
so that it would be evident that DifxApp is deprecated for 5 years

**Comment by wmanning** [comment]
@barnson Maybe this could be moved to a discussion?  It appears only mods have that ability.

We are using difxapp to install a plug and play (USB) driver from a third party (ST Micro) with no issues.  I don't have any experience with kernel drivers. 
We have the following element
`<difx:Driver` AddRemovePrograms="yes" DeleteFiles="no" PlugAndPlayPrompt="no" ForceInstall="no" Legacy="yes" />`
in the <Component> element containing the driver INF file. Legacy is set to yes as we need it to work for Windows XP. Windows XP does not recognize the signed drivers if not signed by Microsoft. Otherwise "no" also works.

**Comment by AbsolutelyWrong** [comment]
@wmanning are you installing only on XP?
Documentation states that  "Starting in Windows 10 Version 1607 (Redstone 1), the Driver Install Frameworks (DIFx) tools (Difxapi.dll, Difxapp.dll, Difxappa.dll, and DPInst.exe) are deprecated and are no longer included in the WDK." this probably has no effect on XP.

**Reply by wmanning** [reply]
XP and above including windows 10.

**Reply by AbsolutelyWrong** [reply]
Have you tracked at which moment Plug and Play Service section is executed in setupapi.dev.log? During driver installation or when associated device is plugged in?

**Reply by wmanning** [reply]
No. I had no need to. To add I'm not a driver writer myself.  ST Micro provides the driver and INF file and we only have to modify a portion of the INF file to customize the strings to our company name and USB PID and VID to our product.  We used a code signing certificate for Windows 8.1 and below and an attestation signed driver or Windows 10.

For some of our newer products we stopped doing that customization and just use the standard ST micro driver as we use ST Micro's standard USB bootloader.
